Logic/src/Messages/PostMessage.cs

```diff
@@ -0,0 +1,103 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Extensions.Logging;
+using System.Text.Json;
+using Microsoft.AspNetCore.Http;
+
+namespace GoRideShare
+{
+    public class PostMessage 
+    {
+        private readonly ILogger<PostMessage> _logger;
+        private readonly string? _baseApiUrl;
+        private readonly IHttpRequestHandler _httpRequestHandler;
+
+        public PostMessage(ILogger<PostMessage> logger, IHttpRequestHandler httpRequestHandler)
+        {
+            _logger = logger;
+            _httpRequestHandler = httpRequestHandler;
+            _baseApiUrl = Environment.GetEnvironmentVariable("BASE_API_URL");
+        }
+
+        // This function is triggered by an HTTP POST request to create a new message in a conversation
+        [Function("PostMessage")]
+        public async Task<IActionResult> Run([HttpTrigger(AuthorizationLevel.Anonymous, "post")] HttpRequest req)
+        {
+            // Read the user ID and the db token from the headers
+            if (!req.Headers.TryGetValue("X-User-ID", out var userId))
+            {
+                return new BadRequestObjectResult("Missing the following header: \'X-User-ID\'.");
+            }
+            if (!req.Headers.TryGetValue("X-Db-Token", out var db_token))
+            {
+                return new BadRequestObjectResult("Missing the following header \'X-Db-Token\'.");
+            }
+
+            // Validate if essential message data is present
```

Comment:
I think it would be helpful for testing and a good practice to move all validation to a separate function. It should be a method that reads the request, if the request is bad, returns an exception. If the request is good, returns null and outputs the request object (Same idea as earlier comment)

This makes the code more modular which makes it easier to read and test.